backend/grading-service/src/services/its/api-wrapper.ts

```diff
@@ -31,41 +33,72 @@ const generateParserString = async (language: string, source_code: string) => {
 
     return parserString;
   } catch (error) {
-    if (error === typeof axios.AxiosError) {
-      throw new ITSPostParserError();
+    if (error instanceof axios.AxiosError) {
```

Comment:
I think you can consider using `axios.isAxiosError()` instead of checking by `instanceof` cause iirc this break OCP/LSP but I'm not sure those apply for this project